### PR TITLE
ReactErrorDecoder: fix using wrong react version's error map

### PR DESCRIPTION
**Pull request by maisymoe**
* make use of `$self` (probably didn't before because old?)
* fetch errors for the *current* React version, not a hardcoded one
* add myself to `Devs` (if my changes aren't fit for this, i can remove)

<hr>

i noticed this (and checked if desktop was the same):
Vendetta debugger, showing a newer React version

given i was already making this PR, i thought i would make a few other small changes too :>

### CODE REVIEW

File: src/plugins/reactErrorDecoder/index.ts [hunk 1]
```diff
@@ -18,27 +18,29 @@
...
                 match: /(function .\(.\)){(for\(var .="https:\/\/reactjs\.org\/docs\/error-decoder\.html\?invariant="\+.,.=1;.<arguments\.length;.\+\+\).\+="&args\[\]="\+encodeURIComponent\(arguments\[.\]\);return"Minified React error #"\+.\+"; visit "\+.\+" for the full message or use the non-minified dev environment for full errors and additional helpful warnings.")}/,
                 replace: (_, func, original) =>
-                    `${func}{var decoded=Vencord.Plugins.plugins.ReactErrorDecoder.decodeError.apply(null, arguments);if(decoded)return decoded;${original}}`,
+                    `${func}{var decoded=$self.decodeError.apply(null, arguments);if(decoded)return decoded;${original}}`,
             },
         },
     ],
 
     async start() {
+        // Moved this into start function because we can't use React top-level, too early
```
Comment (Vendicated) [inline src/plugins/reactErrorDecoder/index.ts:41]:
```suggestion
```

**Review by Cynosphere — APPROVED** [review]

**Review by lexisother — APPROVED** [review]

**Review by ImLvna — APPROVED** [review]

**Review by MrDiamondDog — APPROVED** [review]
hopping on the approval train

**Review by NotNite — APPROVED** [review]
This approval sponsored by the number 39.

**Review by Ruthenic — APPROVED** [review]
LGTM 🚀🚀🚀

**Review by Vendicated — CHANGES_REQUESTED** [review]
looks terrible to me :rocket_crash:

**Review by Vendicated — APPROVED** [review]

**Pull request merged**